### PR TITLE
[JIT] ARM64 - Fix Assertion failed 'node->OperIs(GT_DIV, GT_UDIV, GT_MOD)' during 'Lowering nodeinfo'

### LINKED ISSUE #85630: Assertion failed 'node->OperIs(GT_DIV, GT_UDIV, GT_MOD)' during 'Lowering nodeinfo'
Repro: https://gist.github.com/kunalspathak/46c6f6f196bcf2caf8cbe0ab171d20e0

```
Environment:

set COMPlus_AltJitName=clrjit_universal_arm64_x64.dll
set COMPlus_AltJit=Method23

Assert failure(PID 7372 [0x00001ccc], Thread: 29288 [0x7268]): Assertion failed 'node->OperIs(GT_DIV, GT_UDIV, GT_MOD)' in 'TestClass:Method23(byref,byref,byref,TestClass+S2+S2_D1_F1,byref,byref,float):TestClass+S1:this' during 'Lowering nodeinfo' (IL size 192; hash 0x3cedcacf; FullOpts)
    File: E:\git\runtime\src\coreclr\jit\lowerarmarch.cpp Line: 2117
    Image: e:\git\runtime\artifacts\tests\coreclr\windows.x64.Checked\tests\Core_Root\CoreRun.exe
```

### PR DESCRIPTION
**Pull request by TIHan**
Resolves https://github.com/dotnet/runtime/issues/85630

## Description

There was a case where the node was lowered from `LowerModPow2` but `node->gtNext` would return `nullptr` from `LowerConstIntDivOrMod` which indicates the node was not transformed when it actually was. 

Best way to solve this is return a `bool` indicating if the transformation was successful or not and also provide an out parameter for the next node to lower. So, this renames `LowerConstIntDivOrMod` to `TryLowerConstIntDivOrMod` with an out parameter for the next node to lower.

### CODE REVIEW

**Comment by TIHan** [comment]
@dotnet/jit-contrib @kunalspathak this is ready - [no diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=259546&view=ms.vss-build-web.run-extensions-tab)

File: src/tests/JIT/Regression/JitBlue/Runtime_85630/Runtime_85630.csproj [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Optimize>True</Optimize>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="$(MSBuildProjectName).cs" />
```
Comment (kunalspathak) [inline src/tests/JIT/Regression/JitBlue/Runtime_85630/Runtime_85630.csproj:6]: did you try manually reducing the test? Honestly, this is too big test case to include.

File: src/coreclr/jit/lower.cpp [hunk 2]
```diff
@@ -6596,14 +6596,16 @@ bool Lowering::LowerUnsignedDivOrMod(GenTreeOp* divMod)
...
 // Returns:
-//    nullptr if no transformation is done, or the next node in the transformed node sequence that
-//    needs to be lowered.
+//    false if no transformation is done, true if a transformation is done
 //
-GenTree* Lowering::LowerConstIntDivOrMod(GenTree* node)
+bool Lowering::TryLowerConstIntDivOrMod(GenTree* node, GenTree** nextNode)
 {
     assert((node->OperGet() == GT_DIV) || (node->OperGet() == GT_MOD));
+    assert(nextNode != nullptr);
```
Comment (kunalspathak) [inline src/coreclr/jit/lower.cpp:6607]: Should this be initialized `*nextNode = nullptr` which should be the case if this method returns `false`?

File: src/coreclr/jit/lower.cpp [hunk 3]
```diff
@@ -6871,20 +6877,18 @@ GenTree* Lowering::LowerConstIntDivOrMod(GenTree* node)
...
     if (varTypeIsIntegral(node->TypeGet()))
     {
-        // LowerConstIntDivOrMod will return nullptr if it doesn't transform the node.
-        GenTree* newNode = LowerConstIntDivOrMod(node);
-        if (newNode != nullptr)
+        GenTree* nextNode = nullptr;
+        if (TryLowerConstIntDivOrMod(node, &nextNode))
         {
-            return newNode;
+            return nextNode;
```
Comment (kunalspathak) [inline src/coreclr/jit/lower.cpp:6886]: `assert(nextNode != nullptr)`?

**Review by kunalspathak — CHANGES_REQUESTED** [review]

File: src/tests/JIT/Regression/JitBlue/Runtime_85630/Runtime_85630.csproj (hunk 1 above)
Comment (TIHan) [reply src/tests/JIT/Regression/JitBlue/Runtime_85630/Runtime_85630.csproj:6]: I can try to reduce it down.

File: src/coreclr/jit/lower.cpp (hunk 2 above)
Comment (TIHan) [reply src/coreclr/jit/lower.cpp:6607]: We have a similar `TryLowerAndNegativeOne` with an out parameter `nextNode` and we do not set that to `nullptr` or when we return `false`.

File: src/coreclr/jit/lower.cpp (hunk 3 above)
Comment (TIHan) [reply src/coreclr/jit/lower.cpp:6886]: The issue is that `nextCode` **can** be a `nullptr` even if the transformation was successful.

But, we could add an assert `assert(nextNode == nullptr);` if the transformation wasn't successful.

**Comment by TIHan** [comment]
@kunalspathak this is ready again

**Comment by kunalspathak** [comment]
Surprisingly the simplified test is still not simple, but don't want to waste your time trying to trim it further.

**Review by kunalspathak — APPROVED** [review]
LGTM

**Comment by TIHan** [comment]
Merging. CI failure is unrelated.

**Pull request merged**